prunning_implementation/prunning.py

```diff
@@ -0,0 +1,99 @@
+import babeltrace
+from babeltrace import CTFWriter
+import sys
+import tempfile
+from collections import Counter
+from collections import deque
+
+
+#############################################################################
```

Comment:
I abhor these kinds of comment lines with so much visual clutter. They're not necessary.